pyteal/ast/subroutine.py

```diff
@@ -611,6 +613,29 @@ def method_signature(self, overriding_name: str = None) -> str:
             overriding_name = self.name()
         return f"{overriding_name}({','.join(args)}){self.type_of()}"
 
+    def method_spec(self) -> sdk_abi.Method:
+        skip_names = ["return", "output"]
+
+        args = [
+            {
+                "type": str(abi.type_spec_from_annotation(val)),
+                "name": name,
+            }
+            for name, val in self.subroutine.annotations.items()
+            if name not in skip_names
+        ]
+
+        spec = {
+            "name": self.name(),
+            "args": args,
+            "returns": {"type": str(self.type_of())},
+        }
+
+        if self.subroutine.implementation.__doc__ is not None:
+            spec["desc"] = self.subroutine.implementation.__doc__
```

Comment:
nit: a small improvement for preprocessing `__doc__`:
```python3
spec["desc"] = " ".join([i.strip() for i in spec["desc"].split("\n") if not (i.isspace() or len(i) == 0)])
```
and an example would be
```python3
>>> def func(a):
...     """
...     this is a docstring
...     with breaklines
...     """
...
>>> " ".join([i.strip() for i in func.__doc__.split("\n") if not (i.isspace() or len(i) == 0)])
'this is a docstring with breaklines'
```